_pytest/fixtures.py

```diff
@@ -957,11 +957,7 @@ def __init__(self, session):
 
     def getfixtureinfo(self, node, func, cls, funcargs=True):
         if funcargs and not hasattr(node, "nofuncargs"):
-            if cls is not None:
```

Comment:
Here's what I mean about the "split" responsibility.

Note as well that `getfuncargnames` is an internal utility function used in two places in the code base, it might as well make things simpler for callers to avoid mistakes and keep the calling code clean (as it does here, IMHO).